socorro/cron/jobs/ftpscraper.py

```diff
@@ -296,8 +305,26 @@ def __init__(self, *args, **kwargs):
             HTTPAdapter(max_retries=self.config.retries)
         )
 
+    def url_to_filename(self, url):
+        fn = re.sub('\W', '_', url)
+        fn = re.sub('__', '_', fn)
+        return fn
+
     @memoize_download
     def download(self, url):
+        is_caching = False
+        fn = None
+
+        if url.endswith('.json') and self.config.cachedir:
+            is_caching = True
+            fn = os.path.join(self.config.cachedir, self.url_to_filename(url))
+            if not os.path.isdir(os.path.dirname(fn)):
+                os.makedirs(os.path.dirname(fn))
+            if os.path.exists(fn):
+                with open(fn, 'r') as fp:
+                    print('hit')
+                    return fp.read()
```

Comment:
I tinkered with making this a decorator, but it's tricky since we need the `self.config.cachedir` value. I figured this is good enough for now.